### PR TITLE
Add signalfd support

### PR DESCRIPTION
**Pull request by lun-4**
- Add `os.signalfd` with Linux support. It uses the signalfd4 syscall instead of signalfd as that wasn't working locally. musl uses signalfd4 first and then signalfd as a fallback. I took the liberty of not implementing that, because I'm not sure what is the actual course of action, considering Zig supports Linux 3.16+, and signalfd was added in 2.6.22, with signalfd4 being added in 2.6.27.
 - `signalfd_siginfo` is the result of reading a signalfd. Maybe a helper function to read one would be welcome?
 - Fix a type-related issue I had with `sigaddmask`.

### CODE REVIEW

File: lib/std/os/bits/linux.zig [hunk 1]
```diff
@@ -840,6 +840,31 @@ pub const SIG_IGN = @intToPtr(?Sigaction.sigaction_fn, 1);
 
 pub const empty_sigset = [_]u32{0} ** sigset_t.len;
 
+pub const signalfd_siginfo = extern struct {
+    ssi_signo: u32,
```
Comment (daurnimator) [inline lib/std/os/bits/linux.zig:844]: we usually strip the `ssi_` prefix

File: lib/std/os/bits/linux.zig [hunk 2]
```diff
@@ -840,6 +840,31 @@ pub const SIG_IGN = @intToPtr(?Sigaction.sigaction_fn, 1);
...
+    ssi_trapno: u32,
+    ssi_status: i32,
+    ssi_int: i32,
+    ssi_ptr: u64,
+    ssi_utime: u64,
+    ssi_stime: u64,
+    ssi_addr: u64,
+    ssi_addr_lsb: u16,
+    __pad2: u16,
+    ssi_syscall: i32,
```
Comment (daurnimator) [inline lib/std/os/bits/linux.zig:862]: this should probably be of type `SYS`?
Follow up question: should `SYS` be just 32 bits (currently its a `usize`)?

File: lib/std/os/bits/linux.zig [hunk 3]
```diff
@@ -840,6 +840,31 @@ pub const SIG_IGN = @intToPtr(?Sigaction.sigaction_fn, 1);
...
+    ssi_call_addr: u64,
```
Comment (daurnimator) [inline lib/std/os/bits/linux.zig:863]: odd that this isn't `usize`

Comment (lun-4) [reply lib/std/os/bits/linux.zig:863]: [It is declared as `uint64_t` from musl](https://git.musl-libc.org/cgit/musl/tree/include/sys/signalfd.h#n40), not sure if it should be `usize`

**Pull request merged**